plugin/batch-queue/rm_pmi.cpp

```diff
@@ -138,6 +138,17 @@ extern "C" int PMI_Init( int *spawned )
     return ret;
 }
 
+// Intel MPI uses dlopen() and dlsym() to find pmi functions.
+// We need to redirect PMI_Init() to our wrapper, so that
+// pmi is correctly initialized.
+extern "C" void *dlsym(void *handle, const char *symbol)
+{
+  if (strcmp(symbol, "PMI_Init") == 0) {
```

Comment:
We only need to redirect `PMI_Init()` because it is the first pmi function that gets called. And the rm plugin needs to do some initialization. No other wrappers are required.